### ISSUE TITLE
EMAHandler with warm-up counter-intuitive.

### ISSUE THREAD

**Issue by RicherMans** [pytorch/ignite#2294]
## ❓ Questions/Help/Support

Hello there,
First of all, thanks for the cool package, it's just super useful.

I recently checked out the new version of ignite, which provides a cool new feature: [EMAHandler](https://pytorch.org/ignite/_modules/ignite/handlers/ema_handler.html#EMAHandler).
The EMAHandler uses a momentum value (\alpha), which in contrast to most implementations that I've seen is the inverse decay value (i.e., 1- \alpha = decay).
Usually, when training an EMA model, the online model parameters are unreliable in the first couple epochs, thus in these cases, one might use a high \alpha value of i.e., 0.999, which leads to mainly copying the original parameters over to the EMA model.
Then after some epochs, the EMA models parameters are more reliable and thus \alpha can be set to i.e., 0.001 in order to incrementally update the model.

Thus, in most implementations, there is a linear warmup for the momentum value from i.e., 0.0 to the target decay value (i.e., 0.999).
However, in the current EMAHandler, the warmup momentum (\alpha) is required to be [strictly less than the target momentum](https://github.com/pytorch/ignite/blob/4a37e35dcb9548bb777dc86ab649ff390c6fb631/ignite/handlers/ema_handler.py#L134).
This, therefore leads to only two choices for the momentum: 

1. Start with a high \alpha value and a low (i.e., 0.0) warmup momentum, which just means that at the beginning \theta_{EMA} is used and late into training the online parameters \theta_t.
2. Don't use momentum warmup at all.

In the current EMAHandler implementation, it is impossible to start with a low \alpha (0.0) and increase that value to a high decay (0.999).

My question is, is there something I missed here or why is the implementation using warmup values strictly smaller than the target momentum?

**Comment by sandylaker** [comment]
@vfdev-5 There might be a conflict when using StateParamScheduler to schedule the momentum.

In the current implementation of EMAHandler, we create a state attribute `momentum` when we initialize and attach the handler to engine. But later when we initialize a StateParamScheduler to schedule the `momentum`, it will raise an error due to the fact that `momentum` is already defined in the engine's state.

Please check out [here](https://github.com/pytorch/ignite/blob/79678b916e1fbef702f861d71d5832e17c188351/ignite/handlers/state_param_scheduler.py#L45)

Is there any proper way to overcome this problem?

**Comment by vfdev-5** [comment]
Yes, i opened an issue for that: #2295 
We have to figure out what would be the most reasonable way to fix it...

**Comment by vfdev-5** [comment]
@RicherMans we fixed this issue on master and the fix will be available on conda nightly release in few hours. Could you please check it out and leave us a feedback if everything works as expected for you. Thanks  !

**Comment by RicherMans** [comment]
@vfdev-5 I checked it out yesterday and ran some experiments using EMA. The new version does actually work ( the previous one, as stated above was unable to provide reasonable results).
Thanks a lot for the work!

**Comment by vfdev-5** [comment]
Thanks a lot for the feedback @RicherMans ! We also planning the next release in mid-january 2022 so this fix will be in the stable version
